docs/JSONDocumentation.md

```diff
@@ -0,0 +1,379 @@
+| Parameters            | array of [Parameter](#parameter)                 |                                            |                              |
+| Statements            | array of [Statement](#statement)                 |                                            |                              |
+| Modifiers             | integer, minimum 0                               | See [Modifiers](#modifiers)                | See [Modifiers](#modifiers)  |
+| Attributes            | array of [Attribute](#attribute)                 |                                            | Annotations                  |
+| DocumentationComments | [DocumentationComments](#documentation-comments) | XML documentation comments                 | JavaDoc                      |
+
+
+## Property
+
+Not applicable for Java.
+
+### Required keys:
+
+| Key  | Type   | .NET                 | Java |
+|------|--------|----------------------|------|
+| Name | string | Name of the property | N/A  |
+| Type | string | Type of the property | N/A  |
+
+
+### Optional keys:
+
+| Key                   | Type                                             | .NET                              | Java |
+|-----------------------|--------------------------------------------------|-----------------------------------|------|
+| Initializer           | string                                           | The initial value of the property | N/A  |
+| Modifiers             | integer, minimum 0                               | See [Modifiers](#modifiers)       | N/A  |
+| Attributes            | array of [Attribute](#attribute)                 |                                   | N/A  |
+| DocumentationComments | [DocumentationComments](#documentation-comments) | XML documentation comments        | N/A  |
+
+
+## Attribute
+
+This is a .NET term, it roughly translates to an annotation in Java.
+Note that the arguments in an attribute ([AttributeArgument](#attribute-argument)) hold different properties than [Argument](#argument) as used in other places.
+
+### Required keys:
+
+| Key  | Type   | .NET                  | Java                   |
+|------|--------|-----------------------|------------------------|
+| Name | string | Name of the attribute | Name of the annotation |
+| Type | string | Type of the attribute | Type of the annotation |
+
+
+### Optional keys:
+
+| Key                   | Type                                              | .NET | Java |
+|-----------------------|---------------------------------------------------|------|------|
+| Arguments             | array of [AttributeArgument](#attribute-argument) |      |      |
+
+### Attribute Argument
+
+Argument object specifically designed for the arguments passed with an [Attribute](#attribute).
+
+### Required keys:
+
+| Key   | Type   | .NET                  | Java                  |
+|-------|--------|-----------------------|-----------------------|
+| Name  | string | Name of the argument  | Name of the argument  |
```

Comment:
Java annotation arguments that do not get an explicit name, like this one...
```
@Foo(1234)
class Bar {}
```

implicitly have the name `"value"`. That may be good to mention here.

Reply:
Didn't know that, thanks!